clang/include/clang/CConv/ProgramInfo.h

```diff
@@ -75,12 +74,10 @@ class ProgramInfo : public ProgramVariableAdder {
   CVarSet getVariable(clang::Decl *D, clang::ASTContext *C);
 
   // Retrieve a function's constraints by decl, or by name; nullptr if not found
-  const std::set<FVConstraint *> *getFuncConstraints
-      (FunctionDecl *D, ASTContext *C) const;
-  const std::set<FVConstraint *> *getExtFuncDefnConstraintSet
-      (std::string FuncName) const;
-  const std::set<FVConstraint *> *getStaticFuncConstraintSet
-      (std::string FuncName, std::string FileName) const;
+  FVConstraint *getFuncConstraints (FunctionDecl *D, ASTContext *C) const;
+  FVConstraint *getExtFuncDefnConstraintSet (std::string FuncName) const;
+  FVConstraint *getStaticFuncConstraintSet(std::string FuncName,
+                                           std::string FileName) const;
```

Comment:
Why not returning `const FVConstraint *` ? Does the `const` at the end of the declaration indicate what you want?

Reply:
Maybe the answer is: The thing that is returned can be changed. Before, it was a set, and we insist that the set not be changed, but its contents could be.

Reply:
Yeah, the constraint need to be mutated by the callers at a couple different points. With the set `const` on the return was important ensure that callers could not insert new elements.